api/v3/Domain.php

```diff
@@ -129,8 +133,22 @@ function _civicrm_api3_domain_get_spec(&$params) {
  * @return array
  */
 function civicrm_api3_domain_create($params) {
-  $params['version'] = $params['domain_version'];
-  return _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  if (isset($params['domain_version'])) {
```

Comment:
@johanv This if-else block is good, and it also adds code complexity which should be matched with a test. Can you add a test that changes the domain_version via api?